neuralpredictors/layers/cores/conv2d.py

```diff
@@ -148,14 +148,21 @@ def __init__(
         if depth_separable:
             self.conv_layer_name = "ds_conv"
             self.ConvLayer = DepthSeparableConv2d
+            self.ignore_group_sparsity = True
         elif attention_conv:
             # TODO: check if name attention_conv is backwards compatible
             self.conv_layer_name = "attention_conv"
             self.ConvLayer = self.AttentionConvWrapper
+            self.ignore_group_sparsity = True
         else:
             self.conv_layer_name = "conv"
             self.ConvLayer = nn.Conv2d
+            self.ignore_group_sparsity = False
 
+        if self.ignore_group_sparsity:
+            warnings.warn(
+                "group sparsity can not be calculated for the requested conv type. Hidden channels will not be regularized and gamma_hidden is ignored."
+            )
```

Comment:
I think this warning should only be thrown if a regularization of the hidden channels was actually intended. Otherwise this will be quite annoying and it will also make the user wonder about things they don't need to wonder about. 
I suggest `if self.ignore_group_sparsity and gamma_hidden > 0`

Reply:
Yes, good point. modifying now!